wxGTK31: 3.1.2 -> 3.1.3

evils opened the pull request:
###### Motivation for this change
wxWidgets 3.1.3 has been out since 2019-10-28

###### Things done

<details> <summary> nixpkgs-review </summary>

```
1 package failed to build:
cubicsdr

9 packages built:
diff-pdf grandorgue kicad-unstable kicad-unstable-small prusa-slicer pwsafe treesheets wxGTK31 wxhexeditor
```
</details>

@lasandell `cubicsdr` fails with this update, though comments on their github issues seem to indicate that it should be able to use 3.1.3

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
  - was `179060768` is now `179614160`
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

evils committed:
wxGTK31: 3.1.2 -> 3.1.3

lasandell commented:
The changes here will allow CubicSDR to build with wxGTK 3.1.3: https://github.com/NixOS/nixpkgs/commit/5eb5d023bc90dc09433a22826c12464c6d8b794d. Can you incorporate this into your branch or do I need to open a separate PR?

evils committed:
Allow cubicsdr 0.2.5 to build with wxGTK 3.1.3

evils commented:
@lasandell after a bit of struggling to get that commit onto my branch i found the hashes to mismatch  
i've modified your commit to get the right hashes, added a comment about the source of the patches
and reset authorship so i get the blame for this change

lasandell commented:
Hmm, sorry about that. I used nix-prefetch-url to obtain the hashes, which must be the wrong tool to use for patches.

evils commented:
<details> <summary> nixpkgs-review clears now </summary>

```
10 packages built:
cubicsdr diff-pdf grandorgue kicad-unstable kicad-unstable-small prusa-slicer pwsafe treesheets wxGTK31 wxhexeditor
```
</details>

evils commented:
@peterhoeg, interested in this PR?

peterhoeg commented:
Sure, can you try to fix the merge conflicts please.

evils commented:
merge conflict resolved
<details><summary>nixpkgs-review clears</summary>

```
11 packages built:
cubicsdr diff-pdf grandorgue kicad-unstable kicad-unstable-small prusa-slicer pwsafe treesheets wxGTK31 wxGTK31-gtk3 wxhexeditor
```
</details>

@peterhoeg is there a PR/issue tracking your progress (and intent) for the wxGTK*-gtk3 transition?
it kinda seemed to come out of nowhere...

peterhoeg commented:
>   @peterhoeg is there a PR/issue tracking your progress (and intent) for the
>   wxGTK*-gtk3 transition?
>   it kinda seemed to come out of nowhere...

Not really, no.

It just started because I had to fix it for filezilla some time ago and there are some open PRs to address both the regular transition from gtk2 to gtk3 as well as changing the default for wxgtk.

The way to go about this is to swap wxgtk30/31 to alias -gtk3 and then just specifically use wxgtk30-gtk2 for those that need it.

I know @worldofpeace really wants to do the gtk2 -> gtk3 move, so maybe there are some additional hands there!
